lib/iglu-client/registries.rb

```diff
@@ -41,13 +41,23 @@ def initialize(config, uri)
         @uri = uri
       end
 
-      def lookup_schema(schema_key)
+      def lookup_schema(schema_key, max_retries = 3)
         schema_uri = "#{@uri}/schemas/#{schema_key.as_path}"
+        times_retried = 0
+
         begin
-          response = HTTParty.get(schema_uri)
+          response = HTTParty.get(schema_uri, timeout: 3)
         rescue SocketError => _
```

Comment:
I don't have a strong opinion on that. But leaning towards not retrying on the `SocketError` as this is rarely something that can be fixed by bigger delay.